stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/plugin/listener/SendMessageListenerImpl.kt

```diff
@@ -0,0 +1,47 @@
+package io.getstream.chat.android.offline.experimental.plugin.listener
+
+import android.content.Context
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.experimental.plugin.listeners.SendMessageListener
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.core.ExperimentalStreamChatApi
+import io.getstream.chat.android.offline.experimental.global.GlobalState
+import io.getstream.chat.android.offline.experimental.plugin.logic.LogicRegistry
+import io.getstream.chat.android.offline.message.MessageSendingServiceFactory
+import io.getstream.chat.android.offline.repository.RepositoryFacade
+import kotlinx.coroutines.CoroutineScope
+
+@ExperimentalStreamChatApi
+internal class SendMessageListenerImpl(
+    private val context: Context,
+    private val logic: LogicRegistry,
+    private val globalState: GlobalState,
+    private val scope: CoroutineScope,
+    private val repos: RepositoryFacade,
+) : SendMessageListener {
+
+    override suspend fun onMessageSendResult(
+        result: Result<Message>,
+        channelType: String,
+        channelId: String,
+        message: Message,
+    ) {
+        val service = MessageSendingServiceFactory.getOrCreateService(
```

Comment:
Please don't initialize the dependencies of this class here. The class should not be aware of the initialization of its dependencies. 

This method is basically a delegation to `MessageSendingService` which contains all the logic. You could move the code from  `handleSendMessageSuccess` and `handleSendMessageFail` here as handling the result of the message send it no longer a responsibility of `MessageSendingService` and it is now a responsibility of this class. Then you don't even need to initialize `MessageSendingService` here.

Reply:
Actually, `MessageSendingService::handleSendMessageSuccess` etc are also used by `ChatDomain::sendMessage`. So if I delete `ChatDomain::sendMessage`, then only I can safely remove them from `MessageSendingService`.

Reply:
🤔  True that. In this case, I would copy the method, as this is going to be removed anyway.
Alternatively, you could move `MessageSendingServiceFactory` to `StreamOfflinePluginFactory` and add `MessageSendingService` in the constructor of this listener.